src/token.rs

```diff
@@ -30,77 +26,88 @@ impl Default for Namespace {
     let token = dc_create_id();
-    match chat {
-        Some(chat_id) => context
+    match foreign_id {
+        Some(foreign_id) => context
             .sql
             .execute(
-                "INSERT INTO tokens (namespc, foreign_id, token, timestamp) VALUES (?, ?, ?, ?);",
-                paramsv![namespace, chat_id, token, time()],
+                sqlx::query(
+                    "INSERT INTO tokens (namespc, foreign_id, token, timestamp) VALUES (?, ?, ?, ?);"
+                )
+                    .bind(namespace)
+                    .bind(foreign_id)
+                    .bind(&token)
+                    .bind(time()),
             )
             .await
             .ok(),
         None => context
             .sql
             .execute(
-                "INSERT INTO tokens (namespc, token, timestamp) VALUES (?, ?, ?);",
-                paramsv![namespace, token, time()],
+                sqlx::query(
+                    "INSERT INTO tokens (namespc, token, timestamp) VALUES (?, ?, ?);"
+                )
+                    .bind(namespace)
+                    .bind(&token)
+                    .bind(time()),
             )
             .await
             .ok(),
     };
+
     token
 }
 
 pub async fn lookup(
     context: &Context,
     namespace: Namespace,
     chat: Option<ChatId>,
-) -> Option<String> {
-    match chat {
+) -> crate::sql::Result<Option<String>> {
+    let token = match chat {
         Some(chat_id) => {
             context
                 .sql
-                .query_get_value::<String>(
-                    context,
-                    "SELECT token FROM tokens WHERE namespc=? AND foreign_id=?;",
-                    paramsv![namespace, chat_id],
+                .query_get_value(
+                    sqlx::query("SELECT token FROM tokens WHERE namespc=? AND foreign_id=?;")
+                        .bind(namespace)
+                        .bind(chat_id),
                 )
-                .await
+                .await?
         }
         // foreign_id is declared as `INTEGER DEFAULT 0` in the schema.
         None => {
             context
                 .sql
-                .query_get_value::<String>(
-                    context,
-                    "SELECT token FROM tokens WHERE namespc=? AND foreign_id=0;",
-                    paramsv![namespace],
+                .query_get_value(
+                    sqlx::query("SELECT token FROM tokens WHERE namespc=? AND foreign_id=0;")
+                        .bind(namespace),
                 )
-                .await
+                .await?
         }
-    }
+    };
+    Ok(token)
 }
 
 pub async fn lookup_or_new(
     context: &Context,
     namespace: Namespace,
-    chat: Option<ChatId>,
+    foreign_id: Option<ChatId>,
 ) -> String {
-    if let Some(token) = lookup(context, namespace, chat).await {
+    if let Ok(Some(token)) = lookup(context, namespace, foreign_id).await {
```

Comment:
Errors should be logged?

Reply:
I guess that an error is expected here everytime there is no token yet; in this case, `save` in the line below will be executed